vendor/bat-native-ledger/src/bat_helper.cc

```diff
@@ -2618,6 +2619,25 @@ uint64_t getMediaDuration(const std::map<std::string, std::string>& data,
   return duration;
 }
 
+// static
+bool isValidLeadingUrlSubString(
+    const std::string& url_to_validate,
+    const std::string& to_match) {
+  GURL gurl(url_to_validate);
+  const std::string host_portion =
```

Comment:
I'm confused here, why are we trying to find a host match on something that is not a valid url that contains `://`? ex `.ttvnw.net/v1/segment/`

Reply:
this is a twitch domain that has randomized host names such as: dfafdjk238.ttvnw.net (not really like that, but the host name could come in as anything)

Reply:
The other part of the tertiary is if it doesn't have `://` then start the substr at 0.

Reply:
This has been reworked.